### DISCUSSION TITLE
Update to node-datachannel 0.5.1 (exports fix)

### DISCUSSION THREAD

**Discussion by nasselk** [geckosio/geckos.io#273] [General]
Hey, there were problem with updating to node-datachannel 0.5, but the new version 0.5.1 seems to fix a problem about exports, so maybe it was linked to this issue and should work now (also, since 0.5 node-datachannel dropped support for nodejs versions under 16, and has switched to esm, see : https://github.com/murat-dogan/node-datachannel/issues/201,  but it's more likely due to this exporting problem that was fixed on last release)

![image](https://github.com/geckosio/geckos.io/assets/134150764/10cf1d2b-570a-4053-83fa-206bca421092)

Could you try to update it?

Thanks

**Comment by yandeu** [comment]
Everything works except the for cjs users:
<img width="415" alt="Screenshot 2023-11-19 114701" src="https://github.com/geckosio/geckos.io/assets/20306025/4433a29e-f9bd-4bd9-9e4c-c83c78370493">

I could just remove cjs support and publish it. But there was a long discussions about it and many users want to have cjs support. I think, I should just publish geckos version 3 without cjs support. What do you think? Or is there a easy way to make cjs work? I don't know.

I will quickly update to v0.5.1 and see what happens in the [tests](https://github.com/geckosio/geckos.io/actions/workflows/main.yml).

**Reply by nasselk** [reply]
I personally don't have any problem with switching to version 3 without cjs support since i use esm and ig most of people do.

What about releasing version 3 including node-datachannel v0.5.1 without cjs, and see in the future if node-datachannel re-implement cjs (which the devlopper isnt sure about), and if then re-add it on geckos

**Reply by yandeu** [reply]
Ok. I will release v3 without cjs support.
